elephant/test/test_statistics.py

```diff
@@ -842,6 +846,42 @@ def test_annotations(self):
         self.assertIn('kernel', rate.annotations)
         self.assertEqual(rate.annotations['kernel'], kernel_annotation)
 
+    def test_border_correction(self):
+        np.random.seed(0)
+        n_spiketrains = 75
+        rate = 50. * pq.Hz
+        t_start = 0. * pq.ms
+        t_stop = 1000. * pq.ms
+
+        sampling_period = 0.1 * pq.ms
+
+        trial_list = [homogeneous_poisson_process(
+            rate=rate, t_start=t_start,
+            t_stop=t_stop) for _ in range(n_spiketrains)]
+
+        for CORRECTION in (True, False):
+            rates = []
+            for trial in trial_list:
+                # calculate instaneous rate, discard extra dimension
+                instantenous_rate = statistics.instantaneous_rate(
+                    spiketrains=trial,
+                    sampling_period=sampling_period,
+                    kernel='auto',
+                    border_correction=CORRECTION
+                )
+                rates.append(instantenous_rate)
+            rate_estimated = np.mean(rates, axis=0)[:, 0]
+
+            rtol = 0.05  # Five percent of tolerance
+
+            if CORRECTION:
+                assert np.max(rate_estimated) < (1. + rtol) * rate.item()
+                assert np.min(rate_estimated) > (1. - rtol) * rate.item()
+            else:
+                assert np.max(rate_estimated) < (1. + rtol) * rate.item()
+                # The minimal rate deviates strongly in the uncorrected case.
+                assert not np.min(rate_estimated) > (1. - rtol) * rate.item()
+
```

Comment:
Please use `self.assert*` statements here instead of the regular Python `assert` statement. Otherwise, failing asserts will not be accumulated properly by the test runners.